src/client/shader.cpp

```diff
@@ -730,19 +730,18 @@ ShaderInfo ShaderSource::generateShader(const std::string &name,
 		shaders_header << "#define VOLUMETRIC_LIGHT 1\n";
 	}
 
-	shaders_header << "#line 0\n"; // reset the line counter for meaningful diagnostics
-
 	std::string common_header = shaders_header.str();
+	const char *final_header = "#line 0\n"; // reset the line counter for meaningful diagnostics
```

Comment:
for maximum C++-ness
```suggestion
	std::string_view final_header = "#line 0\n"; // reset the line counter for meaningful diagnostics
```

Reply:
it broke, reverted

```
/mt/src/client/shader.cpp: In member function 'ShaderInfo ShaderSource::generateShader(const std::string&, MaterialType, NodeDrawType)':
/mt/src/client/shader.cpp:740:55: error: no match for 'operator+' (operand types are 'std::__cxx11::basic_string<char>' and 'std::string_view' {aka 'std::basic_string_view<char>'})
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~
      |                                       |                 |
      |                                       |                 std::string_view {aka std::basic_string_view<char>}
      |                                       std::__cxx11::basic_string<char>
In file included from /usr/include/c++/14/string:48,
                 from /usr/include/c++/14/bits/locale_classes.h:40,
                 from /usr/include/c++/14/bits/ios_base.h:41,
                 from /usr/include/c++/14/ios:44,
                 from /usr/include/c++/14/istream:40,
                 from /usr/include/c++/14/fstream:40,
                 from /mt/src/client/shader.cpp:6:
/usr/include/c++/14/bits/stl_iterator.h:627:5: note: candidate: 'template<class _Iterator> constexpr std::reverse_iterator<_Iterator> std::operator+(typename reverse_iterator<_Iterator>::difference_type, const reverse_iterator<_Iterator>&)'
  627 |     operator+(typename reverse_iterator<_Iterator>::difference_type __n,
      |     ^~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:627:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::reverse_iterator<_Iterator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:1797:5: note: candidate: 'template<class _Iterator> constexpr std::move_iterator<_IteratorL> std::operator+(typename move_iterator<_IteratorL>::difference_type, const move_iterator<_IteratorL>&)'
 1797 |     operator+(typename move_iterator<_Iterator>::difference_type __n,
      |     ^~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:1797:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::move_iterator<_IteratorL>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
In file included from /usr/include/c++/14/string:54:
/usr/include/c++/14/bits/basic_string.h:3598:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3598 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3598:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3616:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const _CharT*, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3616 |     operator+(const _CharT* __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3616:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   mismatched types 'const _CharT*' and 'std::__cxx11::basic_string<char>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3635:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(_CharT, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3635 |     operator+(_CharT __lhs, const basic_string<_CharT,_Traits,_Alloc>& __rhs)
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3635:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3652:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, const _CharT*)'
 3652 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3652:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   mismatched types 'const _CharT*' and 'std::basic_string_view<char>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3670:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, _CharT)'
 3670 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs, _CharT __rhs)
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3670:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::basic_string_view<char>')
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3682:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3682 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3682:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3689:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3689 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3689:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3696:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3696 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3696:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3719:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const _CharT*, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3719 |     operator+(const _CharT* __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3719:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   mismatched types 'const _CharT*' and 'std::__cxx11::basic_string<char>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3726:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(_CharT, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3726 |     operator+(_CharT __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3726:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3733:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const _CharT*)'
 3733 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3733:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   mismatched types 'const _CharT*' and 'std::basic_string_view<char>'
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3740:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, _CharT)'
 3740 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3740:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:740:57: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::basic_string_view<char>')
  740 |         vertex_shader = common_header + vertex_header + final_header + vertex_shader;
      |                                                         ^~~~~~~~~~~~
/mt/src/client/shader.cpp:741:59: error: no match for 'operator+' (operand types are 'std::__cxx11::basic_string<char>' and 'std::string_view' {aka 'std::basic_string_view<char>'})
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~
      |                                         |                   |
      |                                         |                   std::string_view {aka std::basic_string_view<char>}
      |                                         std::__cxx11::basic_string<char>
/usr/include/c++/14/bits/stl_iterator.h:627:5: note: candidate: 'template<class _Iterator> constexpr std::reverse_iterator<_Iterator> std::operator+(typename reverse_iterator<_Iterator>::difference_type, const reverse_iterator<_Iterator>&)'
  627 |     operator+(typename reverse_iterator<_Iterator>::difference_type __n,
      |     ^~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:627:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::reverse_iterator<_Iterator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:1797:5: note: candidate: 'template<class _Iterator> constexpr std::move_iterator<_IteratorL> std::operator+(typename move_iterator<_IteratorL>::difference_type, const move_iterator<_IteratorL>&)'
 1797 |     operator+(typename move_iterator<_Iterator>::difference_type __n,
      |     ^~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:1797:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::move_iterator<_IteratorL>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3598:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3598 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3598:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3616:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const _CharT*, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3616 |     operator+(const _CharT* __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3616:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   mismatched types 'const _CharT*' and 'std::__cxx11::basic_string<char>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3635:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(_CharT, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3635 |     operator+(_CharT __lhs, const basic_string<_CharT,_Traits,_Alloc>& __rhs)
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3635:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3652:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, const _CharT*)'
 3652 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3652:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   mismatched types 'const _CharT*' and 'std::basic_string_view<char>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3670:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, _CharT)'
 3670 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs, _CharT __rhs)
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3670:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::basic_string_view<char>')
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3682:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3682 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3682:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3689:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3689 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3689:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3696:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3696 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3696:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3719:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const _CharT*, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3719 |     operator+(const _CharT* __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3719:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   mismatched types 'const _CharT*' and 'std::__cxx11::basic_string<char>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3726:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(_CharT, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3726 |     operator+(_CharT __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3726:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3733:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const _CharT*)'
 3733 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3733:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   mismatched types 'const _CharT*' and 'std::basic_string_view<char>'
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3740:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, _CharT)'
 3740 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3740:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:741:61: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::basic_string_view<char>')
  741 |         fragment_shader = common_header + fragment_header + final_header + fragment_shader;
      |                                                             ^~~~~~~~~~~~
/mt/src/client/shader.cpp:744:67: error: no match for 'operator+' (operand types are 'std::__cxx11::basic_string<char>' and 'std::string_view' {aka 'std::basic_string_view<char>'})
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~
      |                                                 |                   |
      |                                                 |                   std::string_view {aka std::basic_string_view<char>}
      |                                                 std::__cxx11::basic_string<char>
/usr/include/c++/14/bits/stl_iterator.h:627:5: note: candidate: 'template<class _Iterator> constexpr std::reverse_iterator<_Iterator> std::operator+(typename reverse_iterator<_Iterator>::difference_type, const reverse_iterator<_Iterator>&)'
  627 |     operator+(typename reverse_iterator<_Iterator>::difference_type __n,
      |     ^~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:627:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::reverse_iterator<_Iterator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:1797:5: note: candidate: 'template<class _Iterator> constexpr std::move_iterator<_IteratorL> std::operator+(typename move_iterator<_IteratorL>::difference_type, const move_iterator<_IteratorL>&)'
 1797 |     operator+(typename move_iterator<_Iterator>::difference_type __n,
      |     ^~~~~~~~
/usr/include/c++/14/bits/stl_iterator.h:1797:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::move_iterator<_IteratorL>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3598:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3598 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3598:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3616:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const _CharT*, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3616 |     operator+(const _CharT* __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3616:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   mismatched types 'const _CharT*' and 'std::__cxx11::basic_string<char>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3635:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(_CharT, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3635 |     operator+(_CharT __lhs, const basic_string<_CharT,_Traits,_Alloc>& __rhs)
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3635:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3652:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, const _CharT*)'
 3652 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3652:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   mismatched types 'const _CharT*' and 'std::basic_string_view<char>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3670:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, _CharT)'
 3670 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs, _CharT __rhs)
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3670:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::basic_string_view<char>')
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3682:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const __cxx11::basic_string<_CharT, _Traits, _Allocator>&)'
 3682 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3682:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'const std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3689:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const __cxx11::basic_string<_CharT, _Traits, _Allocator>&, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3689 |     operator+(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3689:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3696:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3696 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3696:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3719:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(const _CharT*, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3719 |     operator+(const _CharT* __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3719:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   mismatched types 'const _CharT*' and 'std::__cxx11::basic_string<char>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3726:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(_CharT, __cxx11::basic_string<_CharT, _Traits, _Allocator>&&)'
 3726 |     operator+(_CharT __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3726:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   'std::string_view' {aka 'std::basic_string_view<char>'} is not derived from 'std::__cxx11::basic_string<_CharT, _Traits, _Allocator>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3733:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, const _CharT*)'
 3733 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3733:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   mismatched types 'const _CharT*' and 'std::basic_string_view<char>'
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3740:5: note: candidate: 'template<class _CharT, class _Traits, class _Alloc> std::__cxx11::basic_string<_CharT, _Traits, _Allocator> std::operator+(__cxx11::basic_string<_CharT, _Traits, _Allocator>&&, _CharT)'
 3740 |     operator+(basic_string<_CharT, _Traits, _Alloc>&& __lhs,
      |     ^~~~~~~~
/usr/include/c++/14/bits/basic_string.h:3740:5: note:   template argument deduction/substitution failed:
/mt/src/client/shader.cpp:744:69: note:   deduced conflicting types for parameter '_CharT' ('char' and 'std::basic_string_view<char>')
  744 |                 geometry_shader = common_header + geometry_header + final_header + geometry_shader;
      |                                                                     ^~~~~~~~~~~~
```

Reply:
weird.